src/adagrams.js

```diff
@@ -1,15 +1,155 @@
+  'M' : 2, 'Z' : 1}
+
+  const scoreChart = {
+    1:["A","I","E","O","U","L","N","R","S","T"],
+    2:["D","G"],
+    3:["B","C","M","P"],
+    4:["F","H","V","W","Y"],
+    5:["K"],
+    8:["J","X"],
+    10:["Q","Z"]
+}
+const extraScoreChart = [7, 8, 9, 10]
+
+
 export const drawLetters = () => {
+  let letterList= [];
+  for (let letter in letterQuantityDict){
+    let counter = 0;
+    while (counter < letterQuantityDict[letter]){
+      letterList.push(letter);
+      ++counter;
+    }
+  }
+    let letterBank= _.sampleSize(letterList,10);
+    return letterBank;
+  };
+
   // Implement this method for wave 1
-};
+  export const usesAvailableLetters =(word,lettersInHand)=>{
+  if (!word){
+    return false
+  }
+
+  let letterDict = {}
+  for (let letter of lettersInHand){
+    if (letterDict[letter]){
+      letterDict[letter] += 1;
+    }else{
+      letterDict[letter]= 1;
+    }
+  }
+  for (let letter of word){
+    if(!letterDict[letter]){
+      return false;
+    } else if (letterDict[letter]){
+      letterDict[letter]-=1
+      if (letterDict[letter]<0){
+        console.log(false)
+        return false
+      }
+    }
+  }
+  return true
+}
+
+// export const usesAvailableLetters = (input, lettersInHand) => {
+//   // Implement this method for wave 2
 
-export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
-};
+
+// };
 
 export const scoreWord = (word) => {
   // Implement this method for wave 3
+    if (!word){
+    return 0
+  }
+  let totalScore = 0;
+  let wordUpp = word.toUpperCase();
+  let wordList = wordUpp.split('');
+  for (let letter of wordList){
+    for (let [key, value] of Object.entries(scoreChart)){
+      // console.log(typeof key)
+      if (value.includes(letter)=== true){
+        totalScore += parseInt(key);
+      }
+    }
+  }
+
+  if (extraScoreChart.includes(word.length)){
+    totalScore += 8;
+  }
+  return parseInt(totalScore);
 };
 
+
+
 export const highestScoreFrom = (words) => {
-  // Implement this method for wave 1
+  // Implement this method for wave 4
+  let wordScores = {};
+  let ties = [];
+  for (let word of words){
+    wordScores[word]= scoreWord(word);
+ }
+let scores = Object.values(wordScores);
+let maximum = Math.max(...scores);
+
+for (let [key,value] of Object.entries(wordScores)){
+
+  if (value === maximum){
+    ties.push(key);
+  }
+}
+let longestWord = ties.sort((a,b)=>b.length -a.length);
+console.log(longestWord)
+console.log(longestWord[0])
+
+let shortestWord = longestWord.at(-1);
+console.log(shortestWord)
+console.log(longestWord.at(-1))
+
+
+// console.log(shortestTies)
+if (ties.length === 1){
+  return makeDict(ties[0])
+  }else{
+    console.log(longestWord);
+  for (let item of longestWord){
+    console.log(item)
+    if(item.length === 10){
+      return makeDict(item);
+    }else if (item.length ===shortestWord.length){
+      return makeDict(item);
+    }
+  }
+  }
+};
+
+const makeDict= (word)=>{
```

Comment:
Consider renaming `makeDict` to something a little more descriptive like `formatWinningResults` or `createWinnerObject`